spec/requestReply_at_operation_with-filter.yaml

```diff
@@ -0,0 +1,66 @@
+channels:
+  root:
+    address: '/'
+    messages:
+      pingStatic:
+        $ref: '#/components/messages/pingStatic'
+      pingDynamic:
+        $ref: '#/components/messages/pingDynamic'
+      pong:
+        $ref: '#/components/messages/pong' 
+      dontWork:
+        $ref: '#/components/messages/genericError' 
+  pongDynamic:
+    address: null # will be used from ping.replyAddress
+    messages:
+      pong:
+        $ref: '#/components/messages/pong' 
+      dontWork:
+        $ref: '#/components/messages/genericError'
+
+
+operations:
+  requestStatic:
+    channel:
+      $ref: '#/channels/root'
+    messages:
+      - $ref: '#/channels/root/messages/pingStatic'
+    reply:
+      channel:
+        $ref: '#/channels/root'
+      messages:
+        - $ref: '#/channels/root/messages/pong'
+        - $ref: '#/channels/root/messages/genericError'
+  requestDynamic:
+    channel:
+      $ref: '#/channels/root'
+    messages:
+      - $ref: '#/channels/root/messages/pingDynamic'
+    reply:
+      address: # Attention: that will be applied to the messages of operations[x].channel
+        location: '$message.header#/replyTo'
+      channel:
+        $ref: '#/channels/pongDynamic'
```

Comment:
Sorry for maybe stupid question, I'm not an expert in this, but I'm more taking it logically and looking to see if the specification is clear - in this case, should the dynamic channel and dynamic address be defined together, or can only the address itself be defined, without the channel? 

Also, I don't think so that reading the `address` from `operations[x].messages` perspective will create problems - for me it's very clear looking at the above examples, which strongly change the tone of the specification.

I wonder if a given operation operates on two messages then the user could have the option to define where this address is located for a given message - something like:

```yaml
operations:
  requestDynamic:
    channel:
      $ref: '#/channels/root'
    messages:
      - $ref: '#/channels/root/messages/pingDynamic1'
      - $ref: '#/channels/root/messages/pingDynamic2'
    reply:
      address:
        - message:
            $ref: '#/channels/root/messages/pingDynamic1'
          location: '$message.header#/replyTo1'
        - message:
            $ref: '#/channels/root/messages/pingDynamic2'
          location: '$message.header#/replyTo2'
      channel:
        $ref: '#/channels/pongDynamic'
```

Of course it's only my thinking out loud, we can always create two operations which will operates on two messages separately, but reply will be similar except this dynamic address.